.github/workflows/composite/docker-builder-agw/action.yml

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build Docker Image For AGW
+description: Build a docker image and push it to a registry
+
+inputs:
+  REGISTRY_USERNAME:
+    description: Registry login username
+    required: true
+  REGISTRY_PASSWORD:
+    description: Registry login password
+    required: true
+  REGISTRY:
+    description: Docker registry
+    required: true
+  IMAGE:
+    description: Docker image stream name
+    required: true
+  TAG:
+    description: Docker image stream tag
+    default: latest
+  DOCKERFILE:
+    description: Docker file
+    required: false
+    default: Dockerfile
+  DOCKERCONTEXT:
+    description: Docker context
+    default: .
+  PUSH:
+    description: Push to registry?
+    # See unresolved bug https://github.com/actions/runner/issues/1483
+    default: "true"
+
+outputs:
+  digest:
+    description: Docker image digest
+    value: ${{ steps.build-docker.outputs.digest }}
+
+runs:
+  using: composite
+  steps:
+    - run: echo "Publishing images to ${{ inputs.REGISTRY }}"
+      shell: bash
+
+    - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ inputs.REGISTRY }}
+        username: ${{ inputs.REGISTRY_USERNAME }}
+        password: ${{ inputs.REGISTRY_PASSWORD }}
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Build and push docker image ${{ inputs.IMAGE }}
+      id: build-docker
+      uses: docker/build-push-action@v3
+      with:
+        push: ${{ inputs.PUSH == 'true' && github.token != null }}
+        context: ${{ inputs.DOCKERCONTEXT }}
+        file: ${{ inputs.DOCKERCONTEXT }}/${{ inputs.DOCKERFILE }}
+        tags: ${{ inputs.REGISTRY }}/${{ inputs.IMAGE }}:${{ inputs.TAG }}
+
+    - run: |
```

Comment:
Could we add a `name:` here and then sort the `shell: bash` below that for better readability?